realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -1684,6 +1685,34 @@ public static boolean compactRealm(RealmConfiguration configuration) {
         return BaseRealm.compactRealm(configuration);
     }
 
+    /**
+     * If the Realm is a partially synchronized Realm, fetch and synchronize the objects of a given
+     * object type that match the given query (in string format).
+     *
+     * The results will be returned asynchronously in the callback.
+     *
+     * @param clazz the class to query.
+     * @param query string query.
```

Comment:
I guess we should point to the website doc when it's available (not sure we can mention NSPredicat format since it's usually unknown to Java dev).

`Query` should not be null (passing a null query is like saying I want to sync back the entire Realm, which defies the point of using this method)